tests/unit/s2n_session_ticket_test.c

```diff
@@ -525,8 +537,11 @@ int main(int argc, char **argv)
         uint64_t mock_delay = server_config->decrypt_key_lifetime_in_nanos + server_config->encrypt_decrypt_key_lifetime_in_nanos;
         EXPECT_SUCCESS(s2n_config_set_wall_clock(server_config, mock_nanoseconds_since_epoch, &mock_delay));
 
-        /* Add a second ST key */
-        EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(server_config, ticket_key_name2, s2n_array_len(ticket_key_name2), ticket_key2, s2n_array_len(ticket_key2), 0));
+        /* Add a second session ticket key with an intro time in the past so that the key is immediately valid */
+        POSIX_GUARD(server_config->wall_clock(server_config->sys_clock_ctx, &now));
```

Comment:
Doesn't setting the first stek reference the wall clock? It might be dangerous to set the wall clock here instead of before any logic uses it. You might be changing more than you think.

Reply:
Line 538 sets a wall clock to return mock_delay + current time. On line 541 I'm just using the altered wall clock, I'm not altering it again.

Reply:
🤦